status/views.py

```diff
@@ -57,7 +60,7 @@ def status(request, username=None):
             user = User.objects.get(username=username)
             submissions = submissions.filter(user=user)
         except:
-            raise Http404('User %s Not Found!' % username)
+            raise Http404
```

Comment:
500 should happen when code raises an unhandled exception and results in an internal server error. I am not sure if that suits.